Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -509,6 +509,89 @@ $ adb install UnCrackable-Level1.objection.apk
+
+```
+angler:/ # lsof -p 6233
+COMMAND     PID       USER   FD      TYPE             DEVICE  SIZE/OFF       NODE NAME
+.foobar.c  6233     u0_a97  cwd       DIR                0,1         0          1 /
+.foobar.c  6233     u0_a97  rtd       DIR                0,1         0          1 /
+.foobar.c  6233     u0_a97  txt       REG             259,11     23968        399 /system/bin/app_process64
+.foobar.c  6233     u0_a97  mem   unknown                                         /dev/ashmem/dalvik-main space (region space) (deleted)
+.foobar.c  6233     u0_a97  mem       REG              253,0   2797568    1146914 /data/dalvik-cache/arm64/system@framework@boot.art
+.foobar.c  6233     u0_a97  mem       REG              253,0   1081344    1146915 /data/dalvik-cache/arm64/system@framework@boot-core-libart.art
+.foobar.c  6233     u0_a97  mem       REG              253,0    307200    1146916 /data/dalvik-cache/arm64/system@framework@boot-conscrypt.art
+.foobar.c  6233     u0_a97  mem       REG              253,0    225280    1146917 /data/dalvik-cache/arm64/system@framework@boot-okhttp.art
+....
+```
+
+##### Open Connections
+
+Directory `/proc/pid/net` provides networking information, but the information in this file is not process specific. Instead, it is the same as information as `/proc/net`. There are multiple files present in this directory, of which `tcp`, `tcp6` and `udp` can be considered important from our perspective.
+
+```
+angler:/proc/7254/net # cat tcp
+sl  local_address rem_address   st tx_queue rx_queue tr tm->when retrnsmt   uid  timeout inode
+...
+69: 1101A8C0:BB2F 9A447D4A:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 75412 1 0000000000000000 20 3 19 10 -1
+70: 1101A8C0:917C E3CB3AD8:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 75553 1 0000000000000000 20 3 23 10 -1
+71: 1101A8C0:C1E3 9C187D4A:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 75458 1 0000000000000000 20 3 19 10 -1
+72: 1101A8C0:B4C3 8A827D4A:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 77221 1 0000000000000000 20 3 27 10 -1
+73: 1101A8C0:B4C2 8A827D4A:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 77216 1 0000000000000000 20 3 31 10 -1
+74: 1101A8C0:9B8E 93187D4A:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 74656 1 0000000000000000 20 3 19 10 -1
+75: 1101A8C0:B4C1 8A827D4A:01BB 01 00000000:00000000 00:00000000 00000000 10093        0 74262 1 0000000000000000 20 3 31 10 -1
+...
+```
+
+Another alternative is to use `netstat` command, which also provides information about the network activity for the complete system in a more readable format, and can be easily filtered as per our requirements, for instance for a given pid.
+
+```
+angler:/ # netstat
```

Comment:
Can you give the example with a PID that you are filtering, as mentioned above?

Reply:
sure. I have updated the example.